src/client/datascience/commands/exportCommands.ts

```diff
@@ -63,23 +63,24 @@ export class ExportCommands implements IDisposable {
         this.disposables.push(disposable);
     }
 
+    // The export command as called by the native notebook interface
+    private async nativeNotebookExport(uri: Uri) {
+        const editor = this.notebookProvider.editors.find((item) => this.fs.arePathsSame(item.file, uri));
+
+        if (editor && editor.model) {
+            const interpreter = editor.notebook?.getMatchingInterpreter();
+            return this.export(editor.model, undefined, undefined, interpreter);
+        } else {
+            return this.export(undefined, undefined, undefined, undefined);
+        }
+    }
+
     private async export(
-        modelOrUri: Uri | INotebookModel,
+        model?: INotebookModel,
         exportMethod?: ExportFormat,
         defaultFileName?: string,
         interpreter?: PythonEnvironment
     ) {
-        defaultFileName = typeof defaultFileName === 'string' ? defaultFileName : undefined;
```

Comment:
This export command was hooked up to the editor/title contribution point for native notebooks and shares a signature with the existing export command that the interactive window / old editor still use. However there were some type issues here. The editor/title command populates both a Uri and an object with editor placement info in it. So it was passing something like this:
(URI, { groupId: 1 })

so that groupId object was getting passed in as the defaultFileName optional parameter. Which I think is why this line was here to remove it.

This seemed like an subversion of the type system and the URI | INotebookModel didn't make the most sense as an overload (the two are not related URI is just used to convert to a model. So I added the new nativenotebook export command to avoid the incorrectly typed parameter and the modelOrUri checking code. That code does the correct conversion and calls into export.